tests/commtrack.js

```diff
@@ -124,13 +124,47 @@ define([
 
         it("should create two transfer blocks with the same parent node", function () {
             util.loadXML();
-            util.addQuestion("Transfer", "t1");
-            util.addQuestion("Transfer", "t2");
+            util.addQuestion("Transfer", "t1").p.src.value = "value";
+            util.addQuestion("Transfer", "t2").p.src.value = "value";
             var xml = util.call("createXML"),
                 $xml = $(xml);
             assert.equal($xml.find("transfer").length, 2, xml);
             assert.equal($xml.find("setvalue[ref='/data/transfer[@type=\\'t1\\']/@src']").length, 1, xml);
             assert.equal($xml.find("setvalue[ref='/data/transfer[@type=\\'t2\\']/@src']").length, 1, xml);
         });
+
+        it("transfer question type should omit src when empty", function () {
+            util.loadXML();
+            var trans = util.addQuestion("Transfer", "t1");
+            trans.p.src.value = "something";
+            trans.p.dest.value = "";
+            var xml = util.call("createXML"),
+                $xml = $(xml);
+            assert.strictEqual($xml.find("transfer[type='t1']").attr("src"), "",
+                "unexpected transfer src attribute\n" + xml);
+            assert.isUndefined($xml.find("transfer[type='t1']").attr("dest"),
+                "unexpected transfer dest attribute\n" + xml);
+            assert.equal($xml.find("setvalue[ref='/data/transfer[@type=\\'t1\\']/@src']").length, 1,
+                "unexpected @src setvalue:\n" + xml);
+            assert.equal($xml.find("setvalue[ref='/data/transfer[@type=\\'t1\\']/@dest']").length, 0,
+                "unexpected @dest setvalue:\n" + xml);
+        });
+
+        it("transfer question type should omit dest when empty", function () {
```

Comment:
Both empty is invalid. Either src or dest is required, and vellum displays a validation error if both are blank. Added a test: https://github.com/dimagi/Vellum/commit/8209bd7f941d1c76c9f78cb49ed820a895761595